### ISSUE TITLE
More info to debuf bunder_adjuster

### ISSUE THREAD

**Issue by bhack** [colmap/colmap#2113]
I am running `colmap bundle_adjuster --input_path --output_path`

And I only have in the output

```
==============================================================================
Global bundle adjustment
==============================================================================

Elapsed time: 0.000 [minutes]
```

So there is not extra info to understand what the outcome is.
The only thing I can see is that in output folder `images.txt` all the 3d points ID are set to '-1` and `points3d.txt` is empty.

**Comment by tsattler** [comment]
If this (`colmap bundle_adjuster --input_path --output_path`) is the call to colmap that you make, then you will need to specify an input directory containing data and an output directory where the result should be written to.

**Comment by tsattler** [comment]
Based on that there is no output of the bundle adjustment iterations and that the bundle adjustment takes 0 time, I would expect that there is something wrong with the input model. Does it contain 3D points?

**Comment by bhack** [comment]
This is the pseudo code to prepare camera poses for Colmap export (the 3d points are the same as the one used in solvePnP on each image):

```python
import cv2
import numpy as np
from scipy.spatial.transform import Rotation

def solvepnp_to_colmap(image_points, object_points, camera_matrix, dist_coeffs):
    # Use OpenCV's solvePnP to get rotation and translation vectors
    ret, rvec, tvec = cv2.solvePnP(object_points, image_points, camera_matrix, dist_coeffs)
    
    # Convert rotation vector to rotation matrix
    R, _ = cv2.Rodrigues(rvec)

    # Convert rotation matrix to quaternion using scipy
    rotation = Rotation.from_matrix(R)
    q = rotation.as_quat()
    
    # COLMAP's quaternion representation is (qw, qx, qy, qz)
    quaternion = [q[3], q[0], q[1], q[2]]

    # Return quaternion and translation for COLMAP
    return quaternion, tvec.reshape(-1).tolist()
```

**Comment by bhack** [comment]
I found an issue about axis orientation conversion so probably the points were not visible and now it is ok.
But I still think that we could improve the output in these cases as currently it is very obscure and it doesn't give any hint to debug the pipeline.

**Comment by tsattler** [comment]
Glad to hear that you found the issue.

I am sure @ahojnnes would be glad if you open a pull request for the bundle adjuster to print out more information.

**Comment by bhack** [comment]
Is there a way to run sba fixing 3d points (only poses and intrinsics refinements)?

**Comment by tsattler** [comment]
Yes, it is possible to fix parts of an optimization problem when using ceres.

**Comment by tsattler** [comment]
No, there are only options to fix intrinsics and (part of the) extrinsics.

**Comment by bhack** [comment]
Can we open a new feature request to add this command line flag?

**Comment by tsattler** [comment]
I am sure @ahojnnes would be happy if you implement this and create a pull request.

**Comment by bhack** [comment]
I don't know the colmap internals in the meantime I've opened a feature request. If you have any pinpointer in the code I could evaluate.

**Comment by tsattler** [comment]
The best starting point is to look into colmap.cc to see which functions are called when calling the `bundle_adjuster` and to descend from there into the code.

**Comment by ahojnnes** [comment]
Closing: user found their own axis orientation issue; remaining requests filed separately.
